src/baldr/datetime.cc

```diff
@@ -46,14 +46,20 @@ const tz_db_t& get_tz_db() {
 }
 
 // get a formatted date.  date in the format of 2016-11-06T01:00 or 2016-11-06
-date::local_seconds get_formatted_date(const std::string& date) {
+date::local_seconds get_formatted_date(const std::string& date, bool can_throw) {
   std::istringstream in{date};
   date::local_seconds tp;
 
   if (date.find('T') != std::string::npos)
     in >> date::parse("%FT%R", tp);
   else if (date.find('-') != std::string::npos)
     in >> date::parse("%F", tp);
+  else
+    in.setstate(std::ios::failbit);
+
+  // we weren't able to use this string as a date and you'd like to know about it
+  if (can_throw && in.fail())
+    throw std::invalid_argument("Date string is invalid: " + date);
```

Comment:
@gknisely if the caller says you can throw then we throw when they  pass an invalid date format. the default is to NOT throw which is how the code was previously working. currently only the new code ive written for time tracking asks this to throw. i dont like it but i dont want to go through the exorcise right now of making all the places that call this catch

Reply:
@kevinkreiser That works, but can we create an issue for this with examples of the bad input so that we can update and create tests. Thank you.